sdk/storage/azure-storage-file-datalake/tests/test_file.py

```diff
@@ -907,6 +911,42 @@ def test_rename_file_with_non_used_name(self, datalake_storage_account_name, dat
         self.assertEqual(data, data_bytes)
         self.assertEqual(new_client.path_name, "newname")
 
+    @DataLakePreparer()
+    def test_file_encryption_scope(self, datalake_storage_account_name, datalake_storage_account_key):
+        # Arrange
+        url = self.account_url(datalake_storage_account_name, 'dfs')
+        self.dsc = DataLakeServiceClient(url, credential=datalake_storage_account_key, logging_enable=True)
+        self.config = self.dsc._config
+        self.file_system_name = self.get_resource_name('filesystem')
+        file_name = 'testfile'
+        file_system = self.dsc.get_file_system_client(self.file_system_name)
+        try:
+            file_system.create_file_system(file_system_encryption_scope=TEST_FILE_SYSTEM_ENCRYPTION_KEY_SCOPE)
+        except ResourceExistsError:
+            pass
+
+        file_client = file_system.create_file(file_name)
+        props = file_client.get_file_properties()
+
+        # Assert
+        self.assertTrue(props)
+        self.assertIsNotNone(props['encryption_scope'])
+        self.assertEqual(props['encryption_scope'], TEST_FILE_SYSTEM_ENCRYPTION_KEY_SCOPE.default_encryption_scope)
+
+    @pytest.mark.live_test_only
+    @DataLakePreparer()
+    def test_hns_encryption_scope_sas(self, datalake_storage_account_name, datalake_storage_account_key):
```

Comment:
Three comments here :)

1. This doesn't need to be live only because it doesn't even make a request. Just building a SAS is all done locally.
2. I'm not sure this test belongs in `test_file`. There is nothing specific to a file here.
3. I think we need another test (or two) that actually use an encryption scope SAS to create things and ensure they are created with that encryption scope. For example, use a SAS in a file system client and to create the file system (without explicitly setting the encryption scope) and see if the encryption scope is on the file system. You can do the same thing with a file or directory.

Reply:
1. Good point, I think that was leftover from the original concept of testing it (which did involve creating a File System, but then I got confused looking at the .NET PR)
2. Good point. When I am making the better `SAS` tests, I will also move them to the appropriate files.
3. Will do 😄